edx_proctoring/api.py

```diff
@@ -549,6 +551,76 @@ def get_exam_attempt_by_code(attempt_code):
     return _get_exam_attempt(exam_attempt_obj)
 
 
+def get_exam_attempt_data(exam_id, attempt_id, is_learning_mfe=False):
```

Comment:
we should add unit tests for this function. Since most of this code was just moved over we can probably refactor a lot of the current view tests to test this new function instead.

https://github.com/edx/edx-proctoring/blob/master/edx_proctoring/tests/test_views.py#L1486

Reply:
Can you please explain what's in your mind about refactoring?
My suggestion is that you mean somehow replace `_get_exam_attempt`

Reply:
Because we pulled this logic out of the `StudentProctoredExamAttempt.get` view we should move the tests associated with this logic to unit testing this function now. Otherwise we would have to duplicate all of those test cases for the new `ProctoredExamAttemptView` this PR adds.

Or, we do duplicate all of those tests in our new view but that seems like a comparable amount of work for something would need to be be reworked later to reduce duplication anyway.

Reply:
Thanks for your input. I've moved some of the tests that can be related to newly introduced function and removed them from StudentView tests, please check